### DISCUSSION TITLE
How to store media/static files on Google Cloud Storage (GCS) together with Heroku-deployment

### DISCUSSION THREAD

**Discussion by BrunoVanDamme** [saleor/saleor#7998] [Deployment]
Hi! 

I've recently discovered Saleor and would love to use the project in production. I've managed to deploy: 

1. Saleor (api) to a fresh Heroku-app. The graphql-playground is up and running and works perfectly. I'm using Saleor-release v3.1.0-a.2.

2. Dashboard on Vercel. I'm able to login and edit the products, prices, ... Everything seems to work perfectly (except the images). I'm using Saleor-dashboard release v3.1.0-a.2.

3. Storefront on localhost. I can see the changes made in the dashboard appearing. Everything works fine (except the images). 

I know I have to use S3 or GCS to store my media and static files. I've tried both and none seem to work. I must admit it's the first time I'm using a cloud service from Amazon or Google. I would love to continue with Google, because I like the platform more then AWS. 

I started a fresh bucket on GCS and set the permissions to "public to internet". I've uploaded an image just to test. I've got an public URL to my image: https://storage.googleapis.com/saleor--core/WP_20141007_023.jpg . 

Inside my Heroku app, i've added the key to the media-bucket: 
- Key='GS_MEDIA_BUCKET_NAME' (without quotes)
- Value='saleor--core' (without quotes)

After this I've restarted my Heroku app and saw that the graphql-playground is up and working correctly. However, i'm unable to fetch from the graphql endpoint. It's not working in the dashboard and neither in the storefront. 

I've deleted the GS_MEDIA_BUCKET_NAME key in Heroku and restarted the app. After that, I'm able to fetch from the graphql endpoint from the dashboard and the storefront. 

But my problem with storing the media and static files isn't solved. Does anybody know what i'm doing wrong? It's probably something with the config vars in Heroku? Do I have to add some other keys? Or perhaps there is something wrong with my settings inside the GCS bucket? 

Thanks for pointing me in wright direction. I've been searching for days and can't seem to figure it out.

**Comment by abdullah-bin-hasan** [comment]
Hi, if you are using different buckets for media and static files, you might need to make your objects publicly accessible in both buckets.
also make sure you add all of the required information of your cloud storage such as secret key, access key, endpoint URL etc.
